### DISCUSSION TITLE
CVEs in the keycloak docker image

### DISCUSSION THREAD

**Discussion by wordrak** [keycloak/keycloak#10535] [Archive]
Hello,

our Security Team has found some CVEs in the current version of Keycloak. Do you have a plan to fix them?

```
Vulnerability ID     Severity  Package Name                      Installed Version  Fixed Version    Published Date        Last Modified Date
----------------     --------  ------------                      -----------------  -------------    --------------        ------------------
CVE-2021-23463       CRITICAL  com.h2database:h2                 1.4.197            2.0.202          2021-12-10T20:15:00Z  2022-01-03T21:15:00Z
CVE-2021-42392       CRITICAL  com.h2database:h2                 1.4.197            2.0.206          2022-01-10T14:10:00Z  2022-02-16T16:15:00Z
CVE-2022-23221       CRITICAL  com.h2database:h2                 1.4.197            2.1.210          2022-01-19T17:15:00Z  2022-02-28T22:07:00Z
GMS-2022-7           UNKNOWN   com.h2database:h2                 1.4.197            2.0.206                                
CVE-2021-43859       HIGH      com.thoughtworks.xstream:xstream  1.4.18             1.4.19           2022-02-01T12:15:00Z  2022-02-16T00:15:00Z
CVE-2022-21724       HIGH      org.postgresql:postgresql         42.3.1             42.3.2, 42.2.25  2022-02-02T12:15:00Z  2022-02-07T19:56:00Z
GHSA-673j-qm5f-xpv8  MEDIUM    org.postgresql:postgresql         42.3.1             42.3.3                                 
TOTAL VULNERABILITIES FOUND: 7
CRITICAL/HIGH VULNERABILITIES FOUND: 5
```

**Comment by miguel-carvajal** [comment]
Analized on 19.0.3 and detected:

CVE-2021-3468
CVE-2017-6519
CVE-2019-12900
CVE-2021-25317
CVE-2022-35252
CVE-2022-42010
CVE-2022-42011
CVE-2022-42012
CVE-2020-35512
CVE-2019-8905
CVE-2019-8906
CVE-2022-27404
CVE-2022-27405
CVE-2022-27406
CVE-2018-16428
CVE-2021-43618
CVE-2022-3219
CVE-2022-2509
CVE-2021-4209
CVE-2022-21618
CVE-2022-21626
CVE-2022-21628
CVE-2022-33068
CVE-2022-21619
CVE-2022-21624
CVE-2022-39399
CVE-2020-17049
CVE-2018-16435
CVE-2020-21674
CVE-2017-14166
CVE-2017-14501
CVE-2018-1000879
CVE-2018-1000880
CVE-2022-1304
CVE-2021-42694
CVE-2022-27943
CVE-2018-20657
CVE-2019-14250
CVE-2019-12904
CVE-2019-2201
CVE-2020-13790
CVE-2020-35538
CVE-2022-3515
CVE-2019-7317
CVE-2021-44568
CVE-2018-1000654
CVE-2016-3709
CVE-2022-40303
CVE-2022-40304
CVE-2021-24032
CVE-2019-17543
CVE-2021-39537
CVE-2018-19211
CVE-2018-19217
CVE-2020-12413
CVE-2022-1292
CVE-2018-20225

**Comment by abstractj** [comment]
@wordrak we have been refining as much as we can our security bug triage process to make sure that we fix as much CVEs as we can. Most of the issues are reported here https://github.com/keycloak/keycloak/issues?q=is%3Aissue+sort%3Aupdated-desc+label%3Akind%2Fcve+is%3Aclosed and should be solved in the upcoming release.

If there's anything that we are missing, please do not hesitate to create a GH issue. Adding the CVE ID into the title helps a lot.

@miguel-carvajal almost all the CVEs you mentioned should be solved with a container re-spin, it should happen in the upcoming release, or before.

**Comment by abstractj** [comment]
It should be solved now, at least for the OS dependencies:

```
❯ trivy image --vuln-type os --ignore-unfixed quay.io/keycloak/keycloak:19.0.3
2022-10-26T12:34:42.628-0300    INFO    Detected OS: redhat
2022-10-26T12:34:42.628-0300    INFO    Detecting RHEL/CentOS vulnerabilities...

quay.io/keycloak/keycloak:19.0.3 (redhat 8.6)

Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
```
For Java dependencies, it should be fixed in the upcoming releases.
